types/redis/index.d.ts

```diff
@@ -1175,7 +1176,7 @@ export interface Commands<R> {
 
 export const RedisClient: new (options: ClientOpts) => RedisClient;
 
-export interface RedisClient extends Commands<boolean>, EventEmitter {
+export interface RedisClient<T = boolean> extends Commands<T>, EventEmitter {
```

Comment:
I am not sure if that is what we would want here. It implies that one instance of RedisClient will return the same type for every command which is I think not correct.  
I guess we should either put `any` instead of `boolean` or analyse Redis library and provide a union of types that could be returned. I would vote for `any`. It is done this way in callback version of commands.

Reply:
> I am not sure if that is what we would want here. It implies that one instance of RedisClient will return the same type for every command which is I think not correct.

I agree with @barnski. We should provide the generic for each api. 

> I guess we should either put any.

`unknown` might be better? It isn't supported lower versions though...

Reply:
So either `any` or `unknown`. I agree, `unknown` is better but it will fail the build which tests against older versions for some reason. But I do not know what are the "politics" regarding support for older TS versions.
We could go with `any` since it is used already in callbacks and in root issue the reporter said that callback's behaviour is correct for him.